kerastuner/engine/tuner_utils.py

```diff
@@ -75,145 +75,35 @@ def get_max_epochs_and_steps(fit_args, fit_kwargs):
 
 class TunerCallback(keras.callbacks.Callback):
 
-    def __init__(self, tuner, trial, execution):
+    def __init__(self, tuner, trial):
         self.tuner = tuner
         self.trial = trial
-        self.execution = execution
 
     def on_epoch_begin(self, epoch, logs=None):
         self.tuner.on_epoch_begin(
-            self.execution, self.model, epoch, logs=logs)
+            self.trial, self.model, epoch, logs=logs)
 
     def on_batch_begin(self, batch, logs=None):
-        self.tuner.on_batch_begin(self.execution, self.model, batch, logs)
+        self.tuner.on_batch_begin(self.trial, self.model, batch, logs)
 
     def on_batch_end(self, batch, logs=None):
-        self.tuner.on_batch_end(self.execution, self.model, batch, logs)
+        self.tuner.on_batch_end(self.trial, self.model, batch, logs)
 
     def on_epoch_end(self, epoch, logs=None):
         self.tuner.on_epoch_end(
-            self.execution, self.model, epoch, logs=logs)
+            self.trial, self.model, epoch, logs=logs)
 
 
+# TODO: Add more extensive display.
 class Display(object):
 
-    def __init__(self, host):
```

Comment:
I do really like the look of the display created for KerasTuner. IMO it would be nice if we could bring any relevant parts into core TF Keras. My experience working on the Keras ProgBar is that it's more painful supporting all the options Keras supports than one would think (or hope).
Keras supports a lot of different data types (NumPy, Datasets, generators, Keras Sequence objects) and a lot of different execution modes. Most of these handle steps differently and there's a surprising amount of logic we add to Keras to handle this (One example: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/training_utils.py#L1622)

> When do we not have a single objective?

Internal use cases as well as planned CloudVizier. The `Trial.score` will still be a scalar, but the Oracle's `objective` used to compute that score can be a list of objectives

> Using the Keras progress bar doesn't play nice if you're running in Google Kubernetes Engine - it leads to hundreds of lines of noise

There's a setting `verbose=2` that only outputs 1 line per epoch, this might be what we want here. If that doesn't work, we should have a solution that applies to TF Keras core.

>  It has the Oracle, and the Oracle knows about everything, right?

Theoretically yes, and we can probably at least query the Oracle for remaining_trials. My hope is to keep the Tuner <-> Oracle connection as simple as possible though. Even for something as simple as remaining_trials, some Oracles will likely implement their own stopping logic and so the messaging will be confusing

Reply:
> There's a setting verbose=2 that only outputs 1 line per epoch

That's what I've been adopting, and it seems to work mostly fine for me. 

> My experience working on the Keras ProgBar is that it's more painful supporting all the options Keras supports than one would think (or hope).

This is fair. I'd be fine with agreeing that we should ideally roll what functionality we needed up to the Keras ProgBar, which is realistically just figuring out if

A. We're running in a terminal.  Current ProgBar is fine as is.
B. We're running in an ipython/jupyter/colab setting, in which case, use the ProgBar internal _logic_, but adapt the display logic.
C. If we're running remotely, then just leave it to the user to set the verbosity to 2, and that solves that. 

> My hope is to keep the Tuner <-> Oracle connection as simple as possible though.

We should chat offline a bit on that, because from my thoughts on what it will take to make this work in a distributed setting, my thoughts were leaning towards making the Tuner <-> Oracle connection rather "wide".  I'd much rather sync up, share thoughts, and work out the end plan, than wind up trying to push things in one direction while you're wanting to push in another.